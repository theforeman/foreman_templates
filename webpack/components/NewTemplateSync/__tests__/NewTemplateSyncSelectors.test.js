import { testSelectorsSnapshotWithFixtures } from 'react-redux-test-utils';

import {
  importSettings,
  exportSettings,
  stateFactory,
} from '../__fixtures__/templateSyncSettings.fixtures';

import {
  selectImportSettings,
  selectExportSettings,
  selectLoadingSettings,
  selectError,
} from '../NewTemplateSyncSelectors';

const fixtures = {
  'should return import settings': () =>
    selectImportSettings(stateFactory({ importSettings })),
  'should return export settings': () =>
    selectExportSettings(stateFactory({ exportSettings })),
  'should return loading settings': () =>
    selectLoadingSettings(stateFactory({ loadingSettings: true })),
  'should return loading error': () =>
    selectError(stateFactory({ error: 'Error' })),
};

describe('NewTemplateSyncSelectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
