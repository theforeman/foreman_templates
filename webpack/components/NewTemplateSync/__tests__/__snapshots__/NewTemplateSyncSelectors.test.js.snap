// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTemplateSyncSelectors should return export settings 1`] = `
Array [
  Object {
    "id": 47,
    "name": "filter",
    "settingsType": "string",
    "value": "",
  },
  Object {
    "id": 48,
    "name": "negate",
    "settingsType": "bool",
    "value": false,
  },
  Object {
    "id": 49,
    "name": "repo",
    "settingsType": "string",
    "value": "https://github.com/theforeman/community-templates.git",
  },
]
`;

exports[`NewTemplateSyncSelectors should return import settings 1`] = `
Array [
  Object {
    "id": 45,
    "name": "associate",
    "selection": Array [
      Object {
        "label": "New",
        "value": "new",
      },
      Object {
        "label": "Never",
        "value": "never",
      },
      Object {
        "label": "Always",
        "value": "always",
      },
    ],
    "settingsType": "string",
    "value": "new",
  },
  Object {
    "id": 46,
    "name": "force",
    "settingsType": "bool",
    "value": false,
  },
]
`;

exports[`NewTemplateSyncSelectors should return loading error 1`] = `"Error"`;

exports[`NewTemplateSyncSelectors should return loading settings 1`] = `true`;
